_search-plugins/knn/approximate-knn.md

```diff
@@ -253,7 +253,52 @@ POST _bulk
 ...
 ```
 
-After data is ingested, it can be search just like any other `knn_vector` field!
+After data is ingested, it can be searched just like any other `knn_vector` field.
+
+### Additional query parameters
+
+Starting with version 2.16, k-NN plugin supports additional method parameters to tune search results. This can be done by providing the right values in `method_parameters` section of the query. Here is an example of the query
+
+```json
+GET my-knn-index-1/_search
+{
+  "size": 2,
+  "query": {
+    "knn": {
+      "my_vector2": {
+        "vector": [2, 3, 5, 6],
+        "k": 2,
+        "method_parameters" : {
+          "ef_search": 100
+        }
+      }
+    }
+  }
+}
+```
+These parameters are dependent on the combination of engine and method used to build the index. The tables will help with the valid set of parameters. Invalid combination of parameters will throw an error
+
+#### HNSW
+
+`ef_search`
+
+Explores the `ef_search` number of vectors to find top k nearest neighbors. Higher value will help improve recall at the cost of search latency. The value must be greater than 0
+
+Supported engines | Radial query support | Notes
+:--- | :--- | :---
+Nmslib | no | Overrides the value in index settings if present in search query
+Faiss | yes | Overrides the value in index settings if present in search query
+Lucene | no | Engine supports `k` or `ef_search`. The final result can be controlled by `size`. k-NN plugin will pick `max(k, ef_search)`
```

Comment:
Could you explain this more? Is this saying that users can either specify `k` or `ef_search` but not both? How is `size` affecting the final number of results? Please clarify the last sentence.

Reply:
Not really, Users have to specify `k`. There are validations which will fails otherwise. On engine level, we cannot pass k and ef_search both. If ef_search is present in method parameters it picks the max of both. 

Now as a side effect there will be more results per shard if ef_search value is higher than k. So user will have to use `size` to get the final results. Thinking more about it we can remove the size sentence, since even right now the final results are controlled by `size` if k > size

Reply:
Thanks for the explanation! I think it's helpful to provide the last sentence. Made a suggestion to include this info.